src/docs/arch-prefetch.dox

```diff
@@ -0,0 +1,106 @@
+
+@subsection prefetch_algorithm_find_pages Finding Candidate Pages
+Pre-fetch looks for potential candidate pages in two places: (1) while walking the
+B-tree (\c __tree_walk_internal) and (2) when verifying a tree (\c __verify_tree).
+The \c __wt_session_prefetch_check function contains a few checks to decide if
+pre-fetching should be performed for a given \c ref. Pre-fetching does not operate
+on internal pages, and should not be performed until we have triggered more than
+one leaf page read from sequential read requests. Pre-fetching is also disabled for
+tiered tables, and special operations with the exception of verify.
+
+@subsection prefetch_algorithm_populate_queue Populating the Pre-fetch Queue
+The pre-fetch queue is a first-in-first-out (FIFO) data structure containing the
+page references of potential candidates the system has identified for pre-fetching.
+The \c __wti_btree_prefetch function is responsible for pushing candidate pages
+onto the queue. Given a \c ref, it will queue the next set of pages if they are leaf
+pages, not already in the cache, and not already on the pre-fetch queue. To prevent
+overloading the pre-fetch queue, there is a maximum number of candidate pages allowed
+in the queue (\c WT_MAX_PREFETCH_QUEUE), and a limit on the maximum number of pages
+we can add to the queue during a pre-fetch check (\c WT_PREFETCH_QUEUE_PER_TRIGGER).
+Pages will not be added to the queue if either of these conditions are met.
+
+@subsection prefetch_algorithm_process_pages Pre-fetching Content Into the Cache
+Pre-fetch worker threads are responsible for the work of reading page contents
+into the cache. The \c __wt_prefetch_thread_run function is the entry point for a
+pre-fetch thread, and threads will continue to run while the pre-fetch queue is not
+empty. When there are no more pre-fetch entries to process, pre-fetch worker threads
+wait on a condition variable that signals when an entry has been pushed onto the
+pre-fetch queue. Pre-fetch worker threads invoke the \c __wt_page_in function which
+reads the page from disk and builds an in-memory representation.
+
+@section prefetch_api Pre-fetch API
+@subsection prefetch_api_connection Connection-Level Configuration
+\c wiredtiger_open("prefetch=(available=true,default=false)")
+
+The \c available setting determines whether the pre-fetching functionality can be
+enabled for the given database, and starts up the pool of pre-fetching threads if
+it is turned on. The \c default setting determines whether pre-fetching is turned
+on or off for all sessions belonging to the connection.
+
+@subsection prefetch_api_session Session-Level Configuration
+\c WT_CONNECTION.open_session("prefetch=(enabled=true)")
+
+The session-level configuration can be used to override the connection-level
+configuration for specific sessions if desired.
+
+@section prefetch_other Interaction With Other Components
+
+@subsection prefetch_other_cache_busyness Considering current cache busyness
```

Comment:
Same here, as mentioned in above "@subsection prefetch_algorithm_process_pages Pre-fetching Content Into the Cache".